packages/sdk/utils/src/account.test.ts

```diff
@@ -1,284 +1,499 @@
+      ]
+      const expectedMnemonic = spanishMnemonics[0]
+
+      for (const mnemonic of spanishMnemonics) {
+        expect(normalizeMnemonic(mnemonic)).toEqual(expectedMnemonic)
+      }
+    })
+
+    it('should not normalize accents when the word is from a different language', async () => {
+      // Cases include French mnemonic, missing one accent, and a single Spanish word mixed in, without proper accent.
+      const cases = [
+        {
+          mnemonic:
+            'declarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
+          language: undefined,
+          expected:
+            'déclarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
         },
-      },
-      {
-        derivation0: {
-          address: '0x2bA6d43f0bd3364DD9AF65A80A068a999f311Fd8',
-          privateKey: '4ae237a3fcfcbd877f11cdbf25ea3db398e67c261ddf6448427076da87759b99',
-          publicKey: '03a8c7f50e68fc1c304814ad55837802f0b95b29263cda3d19e86bc26f9eaa5bed',
+        {
+          mnemonic:
+            'declarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
+          language: MnemonicLanguages.french,
+          expected:
+            'déclarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
         },
-        derivation1: {
-          address: '0x773d283771e56abcef5925F9F5Aadb49FB91D647',
-          privateKey: '717dd0b70395964bc4175492f61599e7165cfbc6290fa76798fd77f8dfd9bbb9',
-          publicKey: '0361d8adcac067bb2927d625e642af5f1f53914b102d0740ad97d103ea079a6ce4',
+        {
+          mnemonic:
+            'declarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
+          language: MnemonicLanguages.spanish,
+          expected:
+            'declarer effrayer estime carbone bebé danger déphaser junior buisson ériger morceau cintrer',
         },
-        password: {
-          address: '0xef9212B9803Bcb3CD0aC88De1eC03f617d8DE0e9',
-          privateKey: '89e60bc0273bc04a3f485e1d04d27daeceb0239976176121163c3a94bf3cc150',
-          publicKey: '02e2bdc55d48f3aba547f9da95a30fc3ba54aebb1e80a1ef812679eeba7b84e269',
+      ]
+
+      for (const { mnemonic, language, expected } of cases) {
+        expect(normalizeMnemonic(mnemonic, language)).toEqual(expected)
+      }
+    })
+
+    it('should normalize capitalized words to lowercase', async () => {
+      const mnemonic =
+        'female cousin RAPID exotic ribbon level equiP LeGal fuN RIVER hotel duTy TRIP youth rebel'
+      const expected =
+        'female cousin rapid exotic ribbon level equip legal fun river hotel duty trip youth rebel'
+      expect(normalizeMnemonic(mnemonic)).toEqual(expected)
+    })
+
+    it('should normalize extra and non-standard whitespace', async () => {
```

Comment:
Are there any weird edge cases for whitespace with other languages like Korean or Japanese?

Reply:
Indeed! Only Japanese, but I've added a test case specifically for it now.